cpp/arcticdb/util/error_code.hpp

```diff
@@ -78,14 +78,15 @@ inline std::unordered_map<ErrorCategory, const char*> get_error_category_names()
     ERROR_CODE(5000, E_KEY_NOT_FOUND) \
     ERROR_CODE(5001, E_DUPLICATE_KEY) \
     ERROR_CODE(5002, E_SYMBOL_NOT_FOUND) \
-    ERROR_CODE(5003, E_LMDB_MAP_FULL) \
-    ERROR_CODE(5004, E_PERMISSION) \
-    ERROR_CODE(5005, E_UNEXPECTED_S3_ERROR) \
-    ERROR_CODE(5006, E_S3_RETRYABLE) \
-    ERROR_CODE(5007, E_UNEXPECTED_AZURE_ERROR) \
-    ERROR_CODE(5008, E_UNEXPECTED_ROCKSDB_ERROR) \
-    ERROR_CODE(5009, E_MONGO_BULK_OP_NO_REPLY) \
-    ERROR_CODE(5010, E_UNEXPECTED_MONGO_ERROR) \
+    ERROR_CODE(5003, E_PERMISSION) \
+    ERROR_CODE(5010, E_LMDB_MAP_FULL) \
+    ERROR_CODE(5011, E_UNEXPECTED_LMDB_ERROR) \
+    ERROR_CODE(5020, E_UNEXPECTED_S3_ERROR) \
+    ERROR_CODE(5021, E_S3_RETRYABLE) \
+    ERROR_CODE(5030, E_UNEXPECTED_AZURE_ERROR) \
+    ERROR_CODE(5040, E_UNEXPECTED_ROCKSDB_ERROR) \
+    ERROR_CODE(5050, E_MONGO_BULK_OP_NO_REPLY) \
+    ERROR_CODE(5051, E_UNEXPECTED_MONGO_ERROR) \
```

Comment:
I think this is a good idea (grouping codes for the same storage together). I was going to suggest this on an earlier PR, but since it's (a very minor) API change I was in two minds. Let's make sure we announce the change to error codes in the release notes for the next release.